task_02/nikolai_chernyshov/main.cpp

```diff
@@ -0,0 +1,167 @@
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <memory>
+#include <vector>
+
+using VertexId = int;
+using EdgeId = int;
+
+struct Edge {
+  inline static EdgeId id_max;
+  EdgeId id;
+  VertexId vertex1, vertex2 = -1; // -1 if doesn't exist
+
+  Edge(VertexId vertex1_, VertexId vertex2_ = -1)
+      : id(id_max), vertex1(vertex1_), vertex2(vertex2_) {
+    id_max++;
+  }
+};
+
+struct Vertex {
+  inline static VertexId id_max;
+  VertexId id;
+  std::vector<EdgeId> edges;
+
+  Vertex(EdgeId parent_edge_id, std::vector<EdgeId> edges_ids) : id(id_max) {
```

Comment:
Чем `parent_edge` отличается от общего списка граней? Все грани для вершины одинаковы, будет логичней передавать их все в одной коллекции:
```suggestion
  Vertex(std::vector<EdgeId> edges_ids) : id(id_max) {
```

Reply:
Вектор - тяжелый объект, который лучше предавать по `const` ссылке